cdc/processor.go

```diff
@@ -1028,6 +1064,25 @@ func (p *oldProcessor) sorterConsume(
 				continue
 			}
 
+			for lastResolvedTs > maxLagWithCheckpointTs+lastCheckPointTs {
```

Comment:
Can we ensure the case `case <-checkDoneTicker.C` in L1141 has no starvation when code enters this forloop